cl/audio/feeds.py

```diff
@@ -31,19 +36,30 @@ def items(self, obj):
         """
         Returns a list of items to publish in this feed.
         """
-        with Session() as session:
-            solr = ExtraSolrInterface(
-                settings.SOLR_AUDIO_URL, http_connection=session, mode="r"
-            )
-            params = {
+        if waffle.flag_is_active(obj, "oa-es-deactivate"):
```

Comment:
how is this working? Waffle documentation says the following:

> Flags compare the current [request](https://docs.djangoproject.com/en/dev/topics/http/urls/#how-django-processes-a-request) to their criteria to decide whether they are active.

and the type hinting for the `flag_is_active` method suggests we should pass a request as the first argument:

```python
def flag_is_active(request: HttpRequest, flag_name: str, read_only: bool = False) -> bool | None:
```

is `obj` an instance of HttpRequest? I thought the **obj** argument was an instance of the `Court` model.

Reply:
Yes, this was working because when a request is not passed to the flag, it simply behaves as a `Switch`, ignoring request data like User Groups or Roles.

However, I agree it's better to pass the request. Otherwise, we won't be able to evaluate this section without having to disable the flag for everyone. As admins, we'd want to check this section before flipping it for all users.

The method `get_object` was being overridden in `SearchPodcast`, returning the request but not in `AllJurisdictionsPodcast` or `JurisdictionPodcast`. I've fixed this, returning both the request and the Court as required.

Once ES is in place for OA, we'll need to remove the flags and additional methods added here.

Let me know what do you think now, thanks!